src/reactionsystem.jl

```diff
@@ -323,6 +323,62 @@ function MT.equations(sys::ReactionSystem)
     return eqs
 end
 
+function MT.defaults(sys::ReactionSystem)

```

Comment:
Similar comments to what I wrote for handling `observed`. Can't you just use `get_defaults` for the constraint system and avoid the `rename`? However, you need to handle subsystems too as at:

https://github.com/SciML/ModelingToolkit.jl/blob/ccf4af6336fb1fbabc51e960ea7b488cb054c47c/src/systems/abstractsystem.jl#L493-L503